pkg/controllers/gameserverset/gameserverset_manager.go

```diff
@@ -198,9 +201,87 @@ func (manager *GameServerSetManager) UpdateWorkload() error {
 }
 
 func (manager *GameServerSetManager) SyncPodProbeMarker() error {
+	gss := manager.gameServerSet
+	sqs := gss.Spec.ServiceQualities
+	c := manager.client
+	ctx := context.Background()
+
+	// get ppm
+	ppm := &kruiseV1alpha1.PodProbeMarker{}
+	err := c.Get(ctx, types.NamespacedName{
+		Namespace: gss.GetNamespace(),
+		Name:      gss.GetName(),
+	}, ppm)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			if sqs == nil {
+				return nil
+			}
+			// create ppm
+			return c.Create(ctx, createPpm(gss))
+		}
+		return err
+	}
+
+	// delete ppm
+	if sqs == nil {
+		return c.Delete(ctx, ppm)
+	}
+
+	// update ppm
+	if util.GetHash(gss.Spec.ServiceQualities) != ppm.GetLabels()[gameKruiseV1alpha1.PpmHashKey] {
```

Comment:
Labels are often used for some indexes, so some information that is not related to indexes, but just records controller logic is recommended to be recorded in annotations.